# HG changeset patch
# User Daisuke TAKAGO <takago@neptune.kanazawa-it.ac.jp>
# Date 1290070978 18000
#      Thu Nov 18 04:02:58 2010 -0500
# Node ID be710ed252fff5df59b18e3e6c3cbc217db582fa
# Parent  a117dc8ea1b934ba90f77add3c5698f1383f7440
gl-render.cc (opengl_renderer::draw_line): layer markers in 2D plots

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,13 @@
+2010-11-18  Daisuke TAKAGO  <takago@neptune.kanazawa-it.ac.jp>
+
+	* gl-render.cc (opengl_renderer::draw_line):
+	Layer markers in 2D plots.
+
 2010-11-18  John W. Eaton  <jwe@octave.org>
 
 	Bug # 31689.
 
 	* ov-str-mat.cc (octave_char_matrix_str::save_ascii):
 	Adapt to change in charMatrix::row_as_string function.
 	* DLD-FUNCTIONS/md5sum.cc: New tests.
 
diff --git a/src/gl-render.cc b/src/gl-render.cc
--- a/src/gl-render.cc
+++ b/src/gl-render.cc
@@ -1872,17 +1872,19 @@ opengl_renderer::draw_line (const line::
         fc = props.get_markerfacecolor_rgb ();
 
       init_marker (props.get_marker (), props.get_markersize (),
                    props.get_linewidth ());
 
       for (int i = 0; i < n; i++)
         {
           if (clip[i] == clip_ok)
-            draw_marker (x(i), y(i), (has_z ? z(i) : 0), lc, fc);
+            draw_marker (x(i), y(i),
+                         has_z ? z(i) : static_cast<double> (i) / n,
+                         lc, fc);
         }
 
       end_marker ();
     }
   
   set_clipping (props.is_clipping ());
 }
 
